src/ClimaOcean.jl

```diff
@@ -46,10 +48,62 @@ function __init__(; remove_existing_data=false)
     remove_existing_data && rm(datadep"near_global_quarter_degree", recursive=true, force=true)
 end
 
+struct CubicSplineFunction{d, FT, S} <: Function
+    coordinates :: Vector{FT}
+    nodes :: Vector{FT}
+    spline :: S
+    
+    @doc """
+        CubicSplineFunction{d}(coordinates, nodes, FT=Float64)
+
+    Return a function-like object that interpolates `nodes` between the `coordinates`
+    in dimension `d` using cubic splines. `d` can be either `x`, `y`, or `z`.
+    """
+    function CubicSplineFunction{d}(coordinates, nodes, FT=Float64) where d
+        # Hack to enforce no-gradient boundary conditions,
+        # since CubicSplines doesn't support natively
+        ΔL = coordinates[2] - coordinates[1]
+        pushfirst!(coordinates, coordinates[1] - ΔL)
+
+        ΔR = coordinates[end] - coordinates[end-1]
+        push!(coordinates, coordinates[end] + ΔR)
+    
+        pushfirst!(nodes, nodes[1])
+        push!(nodes, nodes[end])
+    
+        coordinates = Vector{FT}(coordinates)
+        nodes = Vector{FT}(nodes)
+
+        # Now we can build the spline
+        spline = CubicSpline(coordinates, nodes)
+        S = typeof(spline)
+
+        d == :x || d == :y || d == :z || error("Dimension 'd' must be :x or :y or :z")
```

Comment:
would that work with a LatitudeLongitude grid with λ/φ coordinates?

Reply:
yeah, here "x" etc only refers to a positional argument. The only point of using "y" is if you have a two-argument function (such as what we load into `GridFittedBottom`) and need to call this with (x, y), where the spline only varies in the second argument.

We could also get rid of this concept since I don't think it's needed (yet)